lib/Client.js

```diff
@@ -2332,15 +2332,39 @@ class Client extends EventEmitter {
     /**
     * Get the ban list of a guild
     * @arg {String} guildID The ID of the guild
-    * @returns {Promise<Array<Object>>} Resolves with an array of {reason: String, user: User}
-    */
-    getGuildBans(guildID) {
-        return this.requestHandler.request("GET", Endpoints.GUILD_BANS(guildID), true).then((bans) => {
-            bans.forEach((ban) => {
-                ban.user = new User(ban.user, this);
-            });
-            return bans;
+    * @arg {Object} [options] Options for the request
+    * @arg {String} [options.after] Only get users after given user ID
+    * @arg {String} [options.before] Only get users before given user ID
+    * @arg {Number} [options.limit=1000] The maximum number of users to return
+    * @returns {Promise<Array<Object>>} Resolves with an array of { reason: String, user: User }
+    */
+    async getGuildBans(guildID, options = {}) {
+        let bans = await this.requestHandler.request("GET", Endpoints.GUILD_BANS(guildID), true, {
+            after: options.after,
+            before: options.before,
+            limit: options.limit && Math.min(options.limit, 1000)
         });
+
+        for(const ban of bans) {
+            ban.user = this.users.update(ban.user, this);
+        }
+
+        if(options.limit && options.limit > 1000 && bans.length >= 1000) {
+            const page = await this.getGuildBans(guildID, {
+                after: options.before ? undefined : bans[bans.length - 1].user.id,
+                before: options.before ? bans[0].user.id : undefined,
+                limit: options.limit - bans.length
+            });
+
+            if(options.before) {
+                page.push(...bans);
+                bans = page;
```

Comment:
`Array#unshift` exists.

Reply:
we literally.. already talked about that

Reply:
I have great wifi in the train 🚂